arangod/Cluster/ClusterInfo.h

```diff
@@ -788,6 +790,43 @@ class ClusterInfo final {
   Result finishModifyingAnalyzerCoordinator(std::string_view databaseId,
                                             bool restore);
 
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief init metrics state
+  /// try to propose self as leader or understand who is current leader
+  /// no return while no stopping and we don't know who is leader
+  //////////////////////////////////////////////////////////////////////////////
+  void initMetricsState();
+
+  struct [[nodiscard]] MetricsState {
+    std::optional<ServerID> leader;
+    uint64_t version;
+  };
+
+  //////////////////////////////////////////////////////////////////////////////
+  /// @brief get current MetricsState from agencyCache
+  /// @param wantLeader if true we need to propose self to leader if current die
+  /// @note can throw exception if something not founded, or another error
+  /// @note if leader is nullopt that means our server is a leader
+  /// @return who is leader, and what it cache version
```

Comment:
In general I agree your comment better, I will fix it, but

> In particular, this can happen in the bootstrap phase if no MetricsState leader has been set before

this part isn't true. On start every coordinator we go to agency and try to set ourselves as a leader or understand someone another is a leader. So I think only second part about agencyCache is true?

Reply:
[Done?](https://github.com/arangodb/arangodb/pull/16328/commits/a0d579f9d7bf68ac3125604b0039dbacd0f41d5e)